internal/db/utils.go

```diff
@@ -24,3 +35,20 @@ func IsValidSqldUrl(uri string) (bool, string) {
 func EscapeSingleQuotes(value string) string {
 	return strings.Replace(value, "'", "''", -1)
 }
+
+func HasSpecialCharacter(value string) bool {
+	if StartsWithNumber(value) {
+		return true
+	}
+	for _, char := range specialCharacters {
+		if strings.Contains(value, char) {
+			return true
+		}
+	}
+	return false
+}
+
+func StartsWithNumber(value string) bool {
+	firstChar := rune(value[0])
+	return unicode.IsDigit(firstChar)
+}
```

Comment:
Detecting special digits is hard since we can always miss something.
For instance, `+` would be a special character, and it isn't on your list.
I suggest doing the other way around, looking for characters we know we can use without escaping.

Something like:
```go
func NeedsEscaping(name string) bool {
	if len(name) == 0 {
		return false
	}
	for i, char := range name {
		if !unicode.IsLetter(char) && (i == 0 || !unicode.IsDigit(char)) {
			return false
		}
	}
	return true
}
```

In case it needs escaping, then you can wrap it around single quotes and escape any single quotes in the name.

Reply:
I agree with solution, just had to adapt this `NeedsEscaping` a little to make it work. Thanks =)

Reply:
Just need to write some tests (gotta adapt some things)